drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/test/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/AbstractKogitoDMNServiceTest.java

```diff
@@ -171,6 +294,74 @@ public void getFactModelTupleSimpleInputAndDecisionData() {
         assertTrue(decisionDataNameFact.getSimpleProperties().values().contains("string"));
     }
 
+    @Test
+    public void testGetRecursiveDMNTypes() {
```

Comment:
Could we also add a test for nested structures of the same type like
        tPeople[structure]
            nextGroup[tPeople]

Reply:
Well apparently that is not working in BC as well. The Ui in that case makes it impossible to use except by using expressions, so no need for changes here